lib/mysql/mysql.go

```diff
@@ -0,0 +1,651 @@
+	CLIENT_CONNECT_WITH_DB
+	CLIENT_NO_SCHEMA
+	CLIENT_COMPRESS
+	CLIENT_ODBC
+	CLIENT_LOCAL_FILES
+	CLIENT_IGNORE_SPACE
+	CLIENT_PROTOCOL_41
+	CLIENT_INTERACTIVE
+	// CLIENT_SSL = (1<<11)
+	CLIENT_SSL
+	CLIENT_IGNORE_SIGPIPE
+	CLIENT_TRANSACTIONS
+	CLIENT_RESERVED
+	// CLIENT_SECURE_CONNECTION = (1 << 15)
+	CLIENT_SECURE_CONNECTION
+	CLIENT_MULTI_STATEMENTS
+	CLIENT_MULTI_RESULTS
+	CLIENT_PS_MULTI_RESULTS
+	// CLIENT_PLUGIN_AUTH = ( 1 << 19 )
+	CLIENT_PLUGIN_AUTH
+	CLIENT_CONNECT_ATTRS
+	CLIENT_PLUGIN_AUTH_LEN_ENC_CLIENT_DATA
+	CLIENT_CAN_HANDLE_EXPIRED_PASSWORDS
+	CLIENT_SESSION_TRACK
+	CLIENT_DEPRECATED_EOF
+)
+
+// Config defaults
+const (
+	DEFAULT_TIMEOUT_SECS        = 3
+	DEFAULT_PORT                = 3306
+	DEFAULT_CLIENT_CAPABILITIES = CLIENT_SSL
+	DEFAULT_RESERVED_DATA_HEX   = "0000000000000000000000000000000000000000000000"
+)
+
+type Config struct {
+	// @TODO: Does it make sense to make Host/Port connection fields, so that Config can be shared across connections?
+	Host string
+	Port uint16
+
+	TLSConfig          *tls.Config
+	Timeout            time.Duration
+	ClientCapabilities uint32
+	MaxPacketSize      uint32
+	CharSet            byte
+	ReservedData       []byte
+}
+
+// Fill in a (possibly newly-created) Config instance with the default values
+func NewConfig(base *Config) *Config {
+	if base == nil {
+		base = &Config{}
+	}
+	if base.TLSConfig == nil {
+		// @TODO @FIXME Can this be pulled from a global zgrab config module?
+		base.TLSConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+	if base.Port == 0 {
+		base.Port = DEFAULT_PORT
+	}
+	if base.Timeout == 0 {
+		base.Timeout = DEFAULT_TIMEOUT_SECS * time.Second
+	}
+	if base.ClientCapabilities == 0 {
+		base.ClientCapabilities = DEFAULT_CLIENT_CAPABILITIES
+	}
+	if base.ReservedData == nil {
+		bin, err := hex.DecodeString(DEFAULT_RESERVED_DATA_HEX)
+		if err != nil {
+			log.Fatalf("Invalid constant")
+		}
+		base.ReservedData = bin
+	}
+	return base
+}
+
+// Struct holding state for a single connection
+type Connection struct {
+	// Configuration for this connection
+	Config *Config
+
+	// Enum to track connection status
+	State ConnectionState
+	// TCP or TLS-wrapped Connection pointer (IsSecure will tell which)
+	Connection *net.Conn
+	// The sequence number used with the server to number packets
+	SequenceNumber uint8
+	// The "Handshake" packet sent by the server, holding flags used in future calls
+	Handshake *Handshake_Packet
+
+	// If this is true, the Connection is a TLS connection object and there should be a TLSHandshake log.
+	IsSecure bool
+
+	// List of MySQL packets received/sent
+	PacketLog []*PacketLogEntry
+	// The zgrab TLS handshake logs
+	TLSHandshake *tls.ServerHandshake
+}
+
+// Constructor, filling in defaults where needed
+func NewConnection(config *Config) *Connection {
+	return &Connection{
+		Config:         NewConfig(config),
+		State:          STATE_NOT_CONNECTED,
+		PacketLog:      nil,
+		Connection:     nil,
+		SequenceNumber: 0}
```

Comment:
I usually opt for brace on a newline to make it easier to add fields in the future. /shrug
```
    SequenceNumber: 0,
}
```

Reply:
Agree, so much easier to read on a separate line —I remember reading that the trailing comma was an option now, but at the time I was just trying to get rid of the syntax errors. Will address.